proxy/src/project.rs

```diff
@@ -87,3 +87,22 @@ pub fn get(api: &coco::Api, project_urn: &coco::Urn) -> Result<Project, error::E
 
     Ok((project, stats).into())
 }
+
+/// Controversial placeholder struct for `EventStream` projects. It's cumbersome to create
+/// a new `RadUrn` otherwise and the `EventStream` will likely include fields the `Project`
+/// does not (e.g. maintainers)
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct DiscoveryItem {
+    /// Unique identifier of the project in the network.
+    pub id: String,
```

Comment:
We should be adamant to use more concrete types wherever possible. While it introduces more impedence initially the gains over time are outweighing that. It would make @FintanH very happy.

Reply:
@FintanH or @xla what's the best way to create a stubbed `urn` without going through the whole dance of a stubbed `api`?

Reply:
This would make me happy :)

There's an example in `librad`. I'll post it here to save you looking for it this time:

```rust
use librad::{
    hash::Hash,
    uri::{Path, Protocol, RadUrn},
};

let urn = RadUrn::new(
    Hash::hash(b"geez"),
    Protocol::Git,
    Path::parse("rad/issues/42").unwrap(),
);

assert_eq!(
    "rad:git:hwd1yredksthny1hht3bkhtkxakuzfnjxd8dyk364prfkjxe4xpxsww3try/rad/issues/42",
    urn.to_string()
)
```